lib/gcp/firestore.js

```diff
@@ -0,0 +1,83 @@
+"use strict";
+
+var api = require("../../lib/api");
+
+/**
+ * List all collection IDs.
+ *
+ * @param {string} project the Google Cloud project ID.
+ * @return {Promise<string[]>} a promise for an array of collection IDs.
+ */
+var _listCollectionIds = function(project) {
+  var url = "/v1beta1/projects/" + project + "/databases/(default)/documents:listCollectionIds";
```

Comment:
it would be better to have 'v1beta1' as a global variable at the top of the file, makes future migrations easier.